cmd/config-utils.go

```diff
@@ -16,52 +16,45 @@
 
 package cmd
 
-import (
-	"regexp"
-	"strings"
-)
+import "strings"
 
 var validAPIs = []string{"S3v4", "S3v2"}
 
-// isValidSecretKey - validate secret key.
-func isValidSecretKey(secretKey string) bool {
-	if secretKey == "" {
-		return true
-	}
-	regex := regexp.MustCompile(`.{8,40}$`)
-	return regex.MatchString(secretKey) && !strings.ContainsAny(secretKey, "$%^~`!|&*#@")
-}
+const (
+	accessKeyMinLen = 5
+	accessKeyMaxLen = 20
+	secretKeyMinLen = 8
+	secretKeyMaxLen = 40
+)
 
-// isValidAccessKey - validate access key.
+// isValidAccessKey - validate access key for right length.
 func isValidAccessKey(accessKey string) bool {
-	if accessKey == "" {
-		return true
-	}
-	regex := regexp.MustCompile(`.{5,40}$`)
-	return regex.MatchString(accessKey) && !strings.ContainsAny(accessKey, "$%^~`!|&*#@")
+	return len(accessKey) >= accessKeyMinLen && len(accessKey) <= accessKeyMaxLen
+}
+
+// isValidSecretKey - validate secret key for right length.
+func isValidSecretKey(secretKey string) bool {
+	return len(secretKey) >= secretKeyMinLen && len(secretKey) <= secretKeyMaxLen
 }
 
 // isValidHostURL - validate input host url.
-func isValidHostURL(hostURL string) bool {
-	if strings.TrimSpace(hostURL) == "" {
-		return false
-	}
-	url := newClientURL(hostURL)
-	if url.Scheme != "https" && url.Scheme != "http" {
-		return false
-	}
-	if url.Path != "" && url.Path != "/" {
-		return false
+func isValidHostURL(hostURL string) (ok bool) {
```

Comment:
@harshavardhana previous function code flow was more readable. i.e single line indentation and eliminate false conditions one by one.